src/runtime/policy/ingress-validation.ts

```diff
@@ -8,8 +8,56 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {PolicyTarget, Policy} from './policy.js';
-import {Capability} from '../capabilities.js';
+import {assert} from '../../platform/assert-web.js';
+import {PolicyTarget, Policy, PolicyField} from './policy.js';
+import {Capability, Capabilities} from '../capabilities.js';
+import {Type, EntityType} from '../type.js';
+import {Refinement} from '../refiner.js';
+import {Schema} from '../schema.js';
+
+export class IngressValidation {
+  static getRestrictedType(typeName: string, policies: Policy[]): Type|null {
+    const fields = {};
+    let type = null;
+    for (const policy of policies) {
+      for (const target of policy.targets) {
+        if (typeName === target.schemaName) {
+          type = target.type;
+          IngressValidation.mergeFields(fields, target.getRestrictedFields());
```

Comment:
`break;`

And then optionally, after the inner for-loop: `if (type != null) break;` to break out of the outer for-loop too

Reply:
added break to the inner loop. 
i don't want to break out of the outer loop, because i want to merge all types of all relevant policies.